### PR TITLE
Author services added

### LINKED ISSUE #104: AuthorService
#100  needs to be completed before starting this.

Add a AuthorService to the `PubHub.Common` Services folder.

### PR DESCRIPTION
**Pull request by jeongoks**
Closes #104 

- Added the `AuthorService` and added it to the `ServiceExtension` adding the scoped instance.
- Corrected a typo in one of the error responses in the `BookService`.
- Corrected a typo in the `AuthorController` to be singular instead of plural with it returning only one model.

### CODE REVIEW

**Review by ZhakalenDk — APPROVED** [review]

File: PubHub/PubHub.Common/Services/AuthorService.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+                if (authorInfoModel == null)
+                    throw new NullReferenceException($"Unable to map the request over to the client.");
+
+                return authorInfoModel!;
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine("Unable to get authors:", ex.Message);
+                return null!;
```
Comment (Sjimooon) [inline PubHub/PubHub.Common/Services/AuthorService.cs:55]: Wouldn't we want to return an empty list here instead? 🤔 (I know it's been done like this in the other services. I just came to think of it now 🙃)

Comment (ZhakalenDk) [reply PubHub/PubHub.Common/Services/AuthorService.cs:55]: I completely missed this!
Good spot.

We should never return null for a list.
The API itself doesn't even do that.

Comment (jeongoks) [reply PubHub/PubHub.Common/Services/AuthorService.cs:55]: I can change it here and then make an issue to update the other services to do this, if @ZhakalenDk also agrees to do it like this - I am fine with whatever. 😊

Comment (ZhakalenDk) [reply PubHub/PubHub.Common/Services/AuthorService.cs:55]: Absolutely. It's bad practise to rreturn null whne working with collections (:

Comment (jeongoks) [reply PubHub/PubHub.Common/Services/AuthorService.cs]: I changed it - can either one of you review if this is what we want instead? 😊

**Review by Sjimooon — APPROVED** [review]
👍
You could also do `return new();` (or even just `return [];` in .NET 8) when the method return type is a `List<T>`.

**Pull request merged**